expression/builtin_other.go

```diff
@@ -106,11 +137,49 @@ func (c *inFunctionClass) getFunction(ctx sessionctx.Context, args []Expression)
 // builtinInIntSig see https://dev.mysql.com/doc/refman/5.7/en/comparison-operators.html#function_in
 type builtinInIntSig struct {
 	baseBuiltinFunc
+	nonConstArgs []Expression
+	hashSet      map[int64]bool
+	threshold    int
+}
+
+func (b *builtinInIntSig) buildHashMapForConstArgs(ctx sessionctx.Context) error {
+	b.nonConstArgs = make([]Expression, 0, len(b.args))
+	b.nonConstArgs = append(b.nonConstArgs, b.args[0])
+	b.hashSet = make(map[int64]bool, len(b.args)-1)
+	count := 0
+	for i := 1; i < len(b.args); i++ {
+		if b.args[i].ConstItem() {
```

Comment:
I reviewed the meaning of `ConstItem`. It seems that it's not safe to use in planner phase.
You can add some test that the function `IN` meets `prepare && execute` stmt.

Reply:
I have added test case in ` expression/integration_test.go`. It seems nothing goes wrong.